hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -207,6 +215,22 @@ public synchronized int read(final byte[] b, final int off, final int len) throw
     }
     incrementReadOps();
     do {
+      if (nextReadPos >= fCursor - limit && nextReadPos <= fCursor) {
```

Comment:
can you make `fCursor-limit`  a variable with a clear name and use below too, e.g

```
int bytesLeftInBuffer = fCursor -limit;
```